libvirt/resource_libvirt_network.go

```diff
@@ -350,96 +379,27 @@ func resourceLibvirtNetworkCreate(d *schema.ResourceData, meta interface{}) erro
 			// there is no NAT when using a routed network
 			networkDef.Forward.NAT = nil
 		}
+
 		// if addresses are given set dhcp for these
-		err := setDhcpByCIDRAdressesSubnets(d, &networkDef)
+		ips, err := getIPsFromResource(d)
 		if err != nil {
 			return fmt.Errorf("Could not set DHCP from adresses '%s'", err)
 		}
-		var dnsForwarders []libvirtxml.NetworkDNSForwarder
-		if dnsForwardCount, ok := d.GetOk(dnsPrefix + ".forwarders.#"); ok {
-			for i := 0; i < dnsForwardCount.(int); i++ {
-				forward := libvirtxml.NetworkDNSForwarder{}
-				forwardPrefix := fmt.Sprintf(dnsPrefix+".forwarders.%d", i)
-				if address, ok := d.GetOk(forwardPrefix + ".address"); ok {
-					ip := net.ParseIP(address.(string))
-					if ip == nil {
-						return fmt.Errorf("Could not parse address '%s'", address)
-					}
-					forward.Addr = ip.String()
-				}
-				if domain, ok := d.GetOk(forwardPrefix + ".domain"); ok {
-					forward.Domain = domain.(string)
-				}
-				dnsForwarders = append(dnsForwarders, forward)
-			}
-		}
+		networkDef.IPs = ips
 
-		var dnsSRVs []libvirtxml.NetworkDNSSRV
-		if routesCount, ok := d.GetOk(dnsPrefix + ".srvs.#"); ok {
-			for i := 0; i < routesCount.(int); i++ {
-				srv := libvirtxml.NetworkDNSSRV{}
-				srvPrefix := fmt.Sprintf(dnsPrefix+".srvs.%d", i)
-				if service, ok := d.GetOk(srvPrefix + ".service"); ok {
-					srv.Service = service.(string)
-				}
-				if protocol, ok := d.GetOk(srvPrefix + ".protocol"); ok {
-					srv.Protocol = protocol.(string)
-				}
-				if domain, ok := d.GetOk(srvPrefix + ".domain"); ok {
-					srv.Domain = domain.(string)
-				}
-				if target, ok := d.GetOk(srvPrefix + ".target"); ok {
-					srv.Target = target.(string)
-				}
-				if port, ok := d.GetOk(srvPrefix + ".port"); ok {
-					p, err := strconv.Atoi(port.(string))
-					if err != nil {
-						return fmt.Errorf("Could not convert port '%s' to int", port)
-					}
-					srv.Port = uint(p)
-				}
-				if weight, ok := d.GetOk(srvPrefix + ".weight"); ok {
-					w, err := strconv.Atoi(weight.(string))
-					if err != nil {
-						return fmt.Errorf("Could not convert weight '%s' to int", weight)
-					}
-					srv.Weight = uint(w)
-				}
-				if priority, ok := d.GetOk(srvPrefix + ".priority"); ok {
-					w, err := strconv.Atoi(priority.(string))
-					if err != nil {
-						return fmt.Errorf("Could not convert priority '%s' to int", priority)
-					}
-					srv.Priority = uint(w)
-				}
-				dnsSRVs = append(dnsSRVs, srv)
-			}
+		dnsForwarders, err := getDNSForwardersFromResource(d)
```

Comment:
:racing_car: :racing_car: :racehorse:  this is FRESH AIR FOR BRAIN! :brazil: 

we indeed need such refactors, where we have less `noise` in code